### ISSUE TITLE
setupTerm: Couldn't look up terminfo entry "xterm-256color"

### ISSUE THREAD

**Issue by Javran** [jtdaugherty/vty#162]
Environment:
- gentoo linux
- sys-libs/ncurses-5.9-r101 (I created a symbolic link from `/usr/lib64/libtinfo.so.5` to `/usr/lib64/libtinfo.so` to avoid a build time error, see below)
- stackage snapshot lts-12.12
- vty-5.21
- in case it matters there are actually two versions of sys-libs/ncurses in my OS:
  + 5.9-r101(5/5)
  + 6.1-r3(0/6)
  + they are supposed to coexist thanks to [subslots](https://wiki.gentoo.org/wiki/Sub-slots_and_Slot-Operators)
- that being said, I have little to none knowledge about ncurses to tell whether creating the libtinfo.so symbolic link is a proper workaround or whether there are cases where these two ncurses installations will conflict.

After `stack install vty`, I tried the following:

```zsh
$ stack exec -- vty-demo
vty-demo: setupTerm: Couldn't look up terminfo entry "xterm-256color"
$ echo $TERM
xterm-256color
$ ls -l /usr/share/terminfo/x/xterm-256color
lrwxrwxrwx 1 root root 41 Jul  3 04:56 /usr/share/terminfo/x/xterm-256color -> ../../../../etc/terminfo/x/xterm-256color
```

I think the problem is not exclusive to `vty-demo` but any package using `vty`.

Regarding the build time error, when I tried to build vty without `/usr/lib64/libtinfo.so`, I got the following message:

```
    [21 of 36] Compiling Graphics.Vty.Input.Loop ( src/Graphics/Vty/Input/Loop.hs, .stack-work/dist/x86_64-linux/Cabal-2.2.0.1/build/Graphics/Vty/Input/Loop.o )
    <command line>: can't load .so/.DLL for: libtinfo.so (libtinfo.so: cannot open shared object file: No such file or directory)
```

**Comment by jtdaugherty** [comment]
I don't know enough about Gentoo to make use of many of the details you've provided, but: have you verified that the target of the symlink `/usr/share/terminfo/x/xterm-256color` indeed exists? As for the build issue, that appears to be a system library issue independent of `vty`, which I suggest you take up with the maintainer of the `terminfo` package: https://github.com/judah/terminfo

**Comment by Javran** [comment]
I have been investigating for a while with anything noticable.
But today I did two things and it ended up fixing the problem:

- compile `sys-libs/ncurses-6` with `USE=tinfo`
- remove everything in `.stack/` except `.stack/global-project`, and re-setup stack again (I used LTS-12.20 but hopefully other snapshot will work

After these steps are done, I installed brick and now its demo works fine for me.

What i suspect is that you'll need to make sure terminfo is installed **after** system lib ncurses (with `tinfo` USE, who sets up `libtinfo.so` properly.

EDIT:

I think it's possible to just force rebuilding terminfo (once `libtinfo.so` is setup properly) to fix the problem instead of nuking stack.

the answer in https://stackoverflow.com/questions/37236892/force-stack-to-rebuild-an-installed-package worth a try.
